packages/flutter_tools/test/commands.shard/hermetic/generate_localizations_test.dart

```diff
@@ -501,4 +502,10 @@ format: true
       throwsToolExit(message: 'Unexpected positional argument "false".')
     );
   });
+
+  testWithoutContext('Empty ARB can be parsed without FormatException', () {
+    final File arbFile = fileSystem.file(fileSystem.path.join('lib', 'l10n', 'app_en.arb'))
+      ..createSync(recursive: true);
+    expect(AppResourceBundle(arbFile), isA<AppResourceBundle>());
+  });
```

Comment:
Can we also write a more end-to-end test that calls the `gen-l10n` command on an empty `app_en.arb` file. I think we should expect it to run without errors.

Reply:
Sure, added a new test and made them as a new group.